src/core/public/core_system.test.ts

```diff
@@ -17,119 +17,23 @@
-  function _MockInjectedMetadataService(this: any) {
-    this.start = jest.fn().mockReturnValue(mockInjectedMetadataStart);
-    return this;
-  }
-);
-jest.mock('./injected_metadata', () => ({
-  InjectedMetadataService: MockInjectedMetadataService,
-}));
-
-const mockFatalErrorsStart = {};
-const MockFatalErrorsService = jest.fn<FatalErrorsService, any>(function _MockFatalErrorsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockFatalErrorsStart);
-  this.add = jest.fn();
-  return this;
-});
-jest.mock('./fatal_errors', () => ({
-  FatalErrorsService: MockFatalErrorsService,
-}));
-
-const mockI18nStart = {};
-const MockI18nService = jest.fn<I18nService, any>(function _MockI18nService(this: any) {
-  this.start = jest.fn().mockReturnValue(mockI18nStart);
-  this.stop = jest.fn();
-  return this;
-});
-jest.mock('./i18n', () => ({
-  I18nService: MockI18nService,
-}));
-
-const mockNotificationStart = {};
-const MockNotificationsService = jest.fn<NotificationsService, any>(
-  function _MockNotificationsService(this: any) {
-    this.start = jest.fn().mockReturnValue(mockNotificationStart);
-    this.add = jest.fn();
-    this.stop = jest.fn();
-    return this;
-  }
-);
-
-jest.mock('./notifications', () => ({
-  NotificationsService: MockNotificationsService,
-}));
-
-const mockHttp = {};
-const MockHttpService = jest.fn<HttpService, any>(function _MockNotificationsService(this: any) {
-  this.start = jest.fn().mockReturnValue(mockHttp);
-  this.stop = jest.fn();
-  return this;
-});
-jest.mock('./http', () => ({
-  HttpService: MockHttpService,
-}));
-
-const mockBasePathStart = {};
-const MockBasePathService = jest.fn<BasePathService, any>(function _MockNotificationsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockBasePathStart);
-  return this;
-});
-jest.mock('./base_path', () => ({
-  BasePathService: MockBasePathService,
-}));
-
-const mockUiSettings = {};
-const MockUiSettingsService = jest.fn<UiSettingsService, any>(function _MockNotificationsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockUiSettings);
-  this.stop = jest.fn();
-  return this;
-});
-jest.mock('./ui_settings', () => ({
-  UiSettingsService: MockUiSettingsService,
-}));
-
-const mockChromeStart = {};
-const MockChromeService = jest.fn<ChromeService, any>(function _MockNotificationsService(
-  this: any
-) {
-  this.start = jest.fn().mockReturnValue(mockChromeStart);
-  this.stop = jest.fn();
-  return this;
-});
-jest.mock('./chrome', () => ({
-  ChromeService: MockChromeService,
-}));
+import {
+  MockBasePathService,
+  mockBasePathStart,
+  MockChromeService,
+  MockFatalErrorsService,
+  mockFatalErrorsStart,
+  mockHttp,
+  MockHttpService,
+  MockI18nService,
+  mockI18nStart,
+  MockInjectedMetadataService,
+  mockInjectedMetadataStart,
+  MockLegacyPlatformService,
+  MockNotificationsService,
+  mockNotificationStart,
+  MockUiSettingsService,
+} from './core_system.mock';
```

Comment:
Can we use `.mocks.ts` instead since there are multiple mocks?

Reply:
We can @spalger . The idea behind `.mock.ts` was basically to follow the same pattern we have for the test jest files' extensions `.test.ts` which are singular but also contains multiple tests. Do you still think we should use `.mocks.ts` in plural?

Reply:
I think using `.mocks.ts` may be confusing in conjunction with the [the convention that's being established](https://github.com/elastic/kibana/issues/31486) for modules providing generic mocks that can be used in all tests. I think a better approach would be:
- Each module provides a standard mock in `mymodule.mock.ts`. This file would not have any `jest.mock` calls, just dummy objects that other tests can use.
- Each test defines its mocks in `mymodule.test.mocks.ts`. This file would import the relevant mocks needed from the module's dependencies (`*.mock.ts`) and call `jest.mock` for each of them.
- Each test would import its mocks from the mocks file. `mymodule.test.ts` has an import like: `import * as Mocks from './mymodule.test.mocks`.

Reply:
@joshdover I think it is a really good idea! I was not aware of that new convention by the platform team, thanks for letting me know!

Reply:
@spalger @joshdover I just have migrated all the jest mocks to `.test.mocks.js`